### PR TITLE
include token in tsb registration

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/15875

and 
https://bugzilla.redhat.com/show_bug.cgi?id=1470628

### CODE REVIEW

**Comment by bparees** [comment]
will clean this up after https://github.com/openshift/origin/pull/15993 merges

File: install/templateservicebroker/apiserver-template.yaml [hunk 1]
```diff
@@ -134,8 +121,8 @@ objects:
       kubernetes.io/service-account.name: templateservicebroker-client
   type: kubernetes.io/service-account-token
 
-# allow the kube service catalog's SA to read the secret defined above,
-# which will contain the token for the SA that can call the TSB.
+# allow the kube service catalog's SA to read the static secret defined
+# above, which will contain the token for the SA that can call the TSB.
 - apiVersion: authorization.openshift.io/v1
   kind: Role
```
Comment (deads2k) [inline install/templateservicebroker/apiserver-template.yaml]: missed thsi one.

Comment (bparees) [reply install/templateservicebroker/apiserver-template.yaml]: fixed

**Comment by bparees** [comment]
@openshift/devex @deads2k ok this should be good for a final signoff.

File: pkg/oc/bootstrap/docker/openshift/templateservicebroker.go [hunk 2]
```diff
@@ -76,5 +70,65 @@ func (h *Helper) InstallTemplateServiceBroker(f *clientcmd.Factory, imageFormat
 		return errors.NewError(fmt.Sprintf("failed to start the template service broker apiserver: %v", err))
 	}
 
+	// Register the template broker with the service catalog
+	glog.V(2).Infof("registering the template broker with the service catalog")
```
Comment (deads2k) [inline pkg/oc/bootstrap/docker/openshift/templateservicebroker.go]: followup issue to make this a template somewhere please

Comment (bparees) [reply pkg/oc/bootstrap/docker/openshift/templateservicebroker.go]: done: https://github.com/openshift/origin/issues/16075

File: pkg/oc/bootstrap/docker/up.go [hunk 3]
```diff
@@ -408,16 +410,16 @@ func (c *ClientStartConfig) Complete(f *osclientcmd.Factory, cmd *cobra.Command)
...
-	}))
-
 	// Install service catalog
 	c.addTask(conditionalTask("Installing service catalog", c.InstallServiceCatalog, func() bool {
 		return c.ShouldInstallServiceCatalog && c.ShouldInitializeData()
 	}))
 
+	// Install template service broker if our version is high enough
+	c.addTask(conditionalTask("Installing template service broker", c.InstallTemplateServiceBroker, func() bool {
+		return c.ShouldInstallServiceCatalog && c.ShouldInstallTemplateServiceBroker() && c.ShouldInitializeData()
```
Comment (bparees) [inline pkg/oc/bootstrap/docker/up.go]: @csrwng we weren't checking this before, but now we will.

**Pull request merged**